app/assets/javascripts/discourse/app/components/composer/toolbar-buttons.gjs

```diff
@@ -46,7 +56,14 @@ export default class ComposerToolbarButtons extends Component {
               tabindex={{this.tabIndex button}}
               {{on "keydown" this.rovingButtonBar}}
             >
-              {{icon button.icon}}
+              {{#if (this.getLabel button)}}
+                <span title={{this.getLabel button}}>
+                  {{this.getLabel button}}
+                </span>
+              {{/if}}
+              {{#if (this.getIcon button)}}
+                {{icon (this.getIcon button)}}
+              {{/if}}
```

Comment:
I don't know if we shouldn't  use let here to avoid doing two times the same operation for each button, which is like 20 calls in total for a full toolbar, nothing super bad in the big scheme of things... but that's a simple change